lib/wibox/container/arcchart.lua

```diff
@@ -0,0 +1,317 @@
+-- @tparam table children A table composed of valid widgets
+function arcchart:set_children(children)
+    self._private.widget = children and children[1]
+    self:emit_signal("widget::layout_changed")
+end
+
+--- Reset this layout. The widget will be removed and the rotation reset.
+function arcchart:reset()
+    self:set_widget(nil)
+end
+
+for k,v in ipairs {"left", "right", "top", "bottom"} do
+    arcchart["set_"..v.."_padding"] = function(self, val)
+        self._private.paddings = self._private.paddings or {}
+        self._private.paddings[v] = val
+        self:emit_signal("widget::redraw_needed")
+    end
+end
+
+--- The padding between the outline and the progressbar.
+--@DOC_wibox_container_arcchart_padding_EXAMPLE@
+-- @property paddings
+-- @tparam[opt=0] table|number paddings A number or a table
+-- @tparam[opt=0] number paddings.top
+-- @tparam[opt=0] number paddings.bottom
+-- @tparam[opt=0] number paddings.left
+-- @tparam[opt=0] number paddings.right
+
+--- The progressbar value.
+--@DOC_wibox_container_arcchart_value_EXAMPLE@
+-- @property value
+-- @tparam number value Between min_value and max_value
+
+function arcchart:set_value(val)
+    if not val then self._percent = 0; return end
+
+--     if val > 0 then
+--         self:set_max_value(val)
+--     elseif val < 0 then
+--         self:set_min_value(val)
+--     end
+
+    local delta = 0 - 0
+
+    self._percent = val/delta
+    self:emit_signal("widget::redraw_needed")
+end
+
+--- The border background color.
+--@DOC_wibox_container_arcchart_border_color_EXAMPLE@
+-- @property border_color
+
+--- The border foreground color.
+--@DOC_wibox_container_arcchart_color_EXAMPLE@
+-- @property color
+
+--- The border width.
+--@DOC_wibox_container_arcchart_border_width_EXAMPLE@
+-- @property border_width
+-- @tparam[opt=3] number border_width
+
+--- The minimum value.
+-- @property min_value
+
+--- The maximum value.
+-- @property max_value
+
+for _, prop in ipairs {"border_width", "border_color", "paddings", "colors",
+    "rounded_edge", "bg", "thickness", "values", "min_value", "max_value" } do
+    arcchart["set_"..prop] = function(self, value)
+        self._private[prop] = value
+        self:emit_signal("property::"..prop)
+        self:emit_signal("widget::redraw_needed")
+    end
+    arcchart["get_"..prop] = function(self)
+        return self._private[prop] or beautiful["arcchart_"..prop]
+    end
+end
+
+function arcchart:set_paddings(val)
+    self._private.paddings = type(val) == "number" and {
+        left   = val,
+        right  = val,
+        top    = val,
+        bottom = val,
+    } or val or {}
+    self:emit_signal("property::paddings")
+    self:emit_signal("widget::redraw_needed")
+end
+
+--- Returns a new arcchart layout. A arcchart layout arccharts a given widget. Use
+-- :set_widget() to set the widget and :set_direction() for the direction.
+-- The default direction is "north" which doesn't change anything.
```

Comment:
There is no direction on this widget either.

Also, "A arcchart layout archarts a given widget" doesn't really say anything...